src/components/panel/panel.js

```diff
@@ -2539,13 +2548,50 @@ MdPanelPosition.prototype._setPanelPosition = function(panelEl) {
     this._actualPosition = this._positions[i];
     this._calculatePanelPosition(panelEl, this._actualPosition);
     if (this._isOnscreen(panelEl)) {
-      break;
+      return;
     }
   }
+
+  // Class that can be used to re-style the panel if it was repositioned.
+  panelEl.addClass('_md-panel-position-adjusted');
+  this._constrainToViewport(panelEl);
 };
 
 
 /**
+ * Constrains a panel's position to the viewport.
+ * @param {!angular.JQLite} panelEl
+ * @private
+ */
+MdPanelPosition.prototype._constrainToViewport = function(panelEl) {
+  var margin = MdPanelPosition.viewportMargin;
+
+  if (this.getTop()) {
```

Comment:
I doubt it, because the panel is always positioned relatively to the viewport (`position: fixed`). The current datepicker implementation positions the calendar relatively to the body (via `position: absolute`).